src/site/funding/mod.rs

```diff
@@ -1,155 +1,142 @@
-mod icons;
-
 use crate::config::Config;
 use crate::data::funding::{Funding, FundingContent, FundingType};
 use crate::errors::Result;
-use axohtml::dom::UnsafeTextNode;
-use axohtml::elements::{div, li};
-use axohtml::types::SpacedList;
-use axohtml::{html, text, unsafe_text};
-use std::collections::HashMap;
+use serde::Serialize;
+
+#[derive(Serialize, Debug)]
+pub struct FundingContext {
+    preferred_funding: Option<Vec<FundingMethod>>,
+    funding: Vec<FundingMethod>,
+    docs_content: Option<String>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct FundingMethod {
+    title: String,
+    link: String,
+    icon: Option<String>,
+}
 
-/// Generate the standalone funding page.
-pub fn page(config: &Config, funding: &Funding) -> Result<String> {
-    let mut funding_items = funding.content.clone();
-    // We've already made sure that we can unwrap on all of these `Option`s
+pub fn context(config: &Config, funding: &Funding) -> Result<FundingContext> {
+    let mut funding_base = funding.content.clone();
     let unwrapped_config = config.components.funding.as_ref().unwrap();
-    let preferred_html = if unwrapped_config.preferred_funding.is_some() {
+    let preferred_funding = if unwrapped_config.preferred_funding.is_some() {
         let preferred = unwrapped_config.preferred_funding.as_ref().unwrap();
-        // Remove the preferred item from the rest of the list
-        funding_items.remove(preferred);
-        preferred_funding_section(preferred.clone(), funding.content.clone())
+        let content = funding_base.get(preferred).cloned().unwrap();
+        funding_base.remove(preferred);
```

Comment:
Is this unwrap correct? What if the user typo's the key?

Regardless you can replace this with `let content = funding_base.remove(preferred);` (give or take the unwrap)